### PR TITLE
Add support for connector configurations

### PR DESCRIPTION
**Pull request by djaglowski**
Subset of #6700

### CODE REVIEW

**Comment by djaglowski** [comment]
Integration test failure addressed in #6797

**Comment by djaglowski** [comment]
cc @bogdandrutu

File: otelcol/factories.go [hunk 1]
```diff
@@ -16,8 +16,14 @@ package otelcol // import "go.opentelemetry.io/collector/otelcol"
...
 
 // Factories struct holds in a single type all component factories that
 // can be handled by the Config.
-type Factories = component.Factories // nolint:staticcheck
+type Factories struct {
+	component.Factories // nolint:staticcheck
+
+	// Connectors maps connector type names in the config to the respective factory.
+	Connectors map[component.Type]connector.Factory
+}
```
Comment (bogdandrutu) [inline otelcol/factories.go:29]: We may have to wait to do this until next week. now `otelcol.Factories` is a different type than `component.Factories` so this is a breaking change for `otelcol.CollectorSettings.Factories` element.

Comment (djaglowski) [reply otelcol/factories.go:29]: Alright, makes sense.

Comment (bogdandrutu) [reply otelcol/factories.go:29]: This is unblocked if you rebase

Comment (djaglowski) [reply otelcol/factories.go]: I've rebased to a tentative state, with the assumption that #6867 will provide the builder.

File: cmd/builder/internal/config/default.yaml [hunk 2]
```diff
@@ -17,4 +17,6 @@ extensions:
 processors:
   - gomod: go.opentelemetry.io/collector/processor/batchprocessor v0.68.0
   - gomod: go.opentelemetry.io/collector/processor/memorylimiterprocessor v0.68.0
-
+connectors:
+  - import: go.opentelemetry.io/collector/connector/forwardconnector
+    gomod: go.opentelemetry.io/collector/connector/forwardconnector v0.68.0
```
Comment (Aneurysm9) [inline cmd/builder/internal/config/default.yaml:22]: Is the `import` entry needed here?  If the package import path and module identifier are equal I think you only need `gomod`.

Comment (djaglowski) [reply cmd/builder/internal/config/default.yaml]: Good call. Fixed.

**Comment by djaglowski** [comment]
Thanks @jpkrohling. I've added a changelog entry and updated the title.

File: cmd/otelcorecol/builder-config.yaml [hunk 3]
```diff
@@ -17,12 +17,16 @@ extensions:
 processors:
   - gomod: go.opentelemetry.io/collector/processor/batchprocessor v0.69.0
   - gomod: go.opentelemetry.io/collector/processor/memorylimiterprocessor v0.69.0
+connectors:
+  - import: go.opentelemetry.io/collector/connector/forwardconnector
+    gomod: go.opentelemetry.io/collector/connector/forwardconnector v0.69.0
```
Comment (TylerHelmuth) [inline cmd/otelcorecol/builder-config.yaml:22]: Can we use only gomod here as well?

**Review by TylerHelmuth — APPROVED** [review]

File: cmd/otelcorecol/builder-config.yaml (hunk 3 above)
Comment (djaglowski) [reply cmd/otelcorecol/builder-config.yaml]: Good catch, thanks. Done.

**Review by kovrus — APPROVED** [review]

**Review by jpkrohling — APPROVED** [review]

**Pull request merged**